src/ess/powder/conversion.py

```diff
@@ -229,8 +237,35 @@ def convert_reduced_to_tof(
     )
 
 
-def convert_monitor_do_wavelength(
-    monitor: MonitorData[RunType, MonitorType],
+def compute_detector_time_of_flight(
+    detector_data: DetectorData[RunType], tof_workflow: TofWorkflow
+) -> TofData[RunType]:
+    wf = tof_workflow.pipeline.copy()
+    wf[time_of_flight.RawData] = detector_data
+    return TofData[RunType](wf.compute(time_of_flight.TofData))
+
+
+def compute_monitor_time_of_flight(
+    monitor: MonitorData[RunType, MonitorType], tof_workflow: TofWorkflow
+) -> TofMonitorData[RunType, MonitorType]:
+    wf = tof_workflow.pipeline.copy()
+    wf.insert(time_of_flight.resample_tof_data)
+    wf[time_of_flight.RawData] = monitor
+    return TofMonitorData[RunType, MonitorType](
+        wf.compute(time_of_flight.ResampledTofData)
+    )
+
+
+def set_monitor_zeros_to_nan(
+    monitor: TofMonitorData[RunType, MonitorType],
+) -> TofMonitorDataZerosToNan[RunType, MonitorType]:
+    inds = monitor.values == 0.0
+    monitor.values[inds] = np.nan
+    return TofMonitorDataZerosToNan[RunType, MonitorType](monitor)
```

Comment:
Looking back at the discussion about granularity, can this be merged with the above provider? Do we need to expose the data without NaNs here? Or should it be part of `convert_monitor_to_wavelength`?

Also, this provider modifies its input which can lead to surprises down the line. Avoiding this would be easiest by merging it into `compute_monitor_time_of_flight`.